src/images/avatar.ts

```diff
@@ -8,6 +9,36 @@ import {Component, Input, ChangeDetectionStrategy} from 'angular2/core';
 export class NglAvatar {
   @Input() src: string = '';
   @Input() alt: string = '';
-  @Input() size: 'x-small' | 'small' | 'medium' | 'large' = 'medium';
-  @Input() type: 'rectangle' | 'circle' = 'rectangle';
+  @Input() size: 'x-small' | 'small' | 'medium' | 'large';
+
+  @Input('size') set setSize(value: 'x-small' | 'small' | 'medium' | 'large') {
+    this.updateClass(this._size, value);
+    this._size = value;
+  }
+
+  @Input('type') set setType(value: 'rectangle' | 'circle') {
+    this.updateClass(this._type, value);
+    this._type = value;
+  }
+
+  private _type: 'rectangle' | 'circle';
+  private _size: 'x-small' | 'small' | 'medium' | 'large';
+
+  constructor(public element: ElementRef, public renderer: Renderer) {
+    renderer.setElementClass(element.nativeElement, 'slds-avatar', true);
+  }
+
+  ngOnInit() {
+    if (!this._type) {
+      this.renderer.setElementClass(this.element.nativeElement, 'slds-avatar--rectangle', true);
+    }
+
+    if (!this._size) {
+      this.renderer.setElementClass(this.element.nativeElement, 'slds-avatar--medium', true);
+    }
+  }
+
+  private updateClass(oldValue: string, newValue: string) {
+    replaceClass(this, `slds-avatar--${oldValue}`, newValue ? `slds-avatar--${newValue}` : '');
+  }
 };
```

Comment:
To be "politically correct" the issue is related more to the `host: [class]` binding, and it's assumption that it's  the only class-binding player :) So event if `ngClass` was binded to something that is recomputed, in order for this to work correct, when `type` or `size` was changing also the `ngClass`  result had to change! 

You can see what I mean with this test:
```js
it('should change the type of the avatar element based on input', testAsync((tcb: TestComponentBuilder) => {
    return createFixture(tcb, `<ngl-avatar [type]="type" src="image1.jpg" [ngClass]="{'custom-class': hasClass}"></ngl-avatar>`).then((fixture) => {
      fixture.componentInstance.hasClass = false;
      fixture.detectChanges();

      const avatar = getAvatarElement(fixture.nativeElement);

      expect(avatar).toHaveCssClass('slds-avatar--circle');
      expect(avatar).not.toHaveCssClass('custom-class');

      fixture.componentInstance.type = 'rectangle';
      fixture.componentInstance.hasClass = true;
      fixture.detectChanges();
      expect(avatar).toHaveCssClass('slds-avatar--rectangle');
      expect(avatar).not.toHaveCssClass('slds-avatar--circle');
      expect(avatar).toHaveCssClass('custom-class');

      fixture.componentInstance.type = 'circle';
      fixture.detectChanges();
      expect(avatar).toHaveCssClass('slds-avatar--circle');
      expect(avatar).toHaveCssClass('custom-class');
    });
  }));
```

Anyway, I also like the short version of doing things, but it seems not to be feasible in this case.